libs/structwalk/walktype_test.go

```diff
@@ -0,0 +1,285 @@
+package structwalk
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/databricks/cli/libs/structdiff/structpath"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWalkTypeNilCallback(t *testing.T) {
+	err := WalkType(reflect.TypeOf(""), nil)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "visit callback must not be nil")
+}
+
+func TestWalkTypeNilType(t *testing.T) {
+	results := make(map[string]reflect.Type)
+	err := WalkType(nil, func(path *structpath.PathNode, typ reflect.Type) {
+		results[path.DynPath()] = typ
+	})
+	require.NoError(t, err)
+	assert.Empty(t, results)
+}
+
+func TestWalkTypeScalarFields(t *testing.T) {
+	type TestStruct struct {
+		StringField string  `json:"string_field"`
+		IntField    int     `json:"int_field"`
+		BoolField   bool    `json:"bool_field"`
+		FloatField  float64 `json:"float_field"`
+	}
+
+	results := make(map[string]reflect.Type)
+	err := WalkType(reflect.TypeOf(TestStruct{}), func(path *structpath.PathNode, typ reflect.Type) {
+		results[path.DynPath()] = typ
```

Comment:
We should also have a `path.DynPattern()` method. You can translate map / slice types to something like `foo.*.[*]` to denote a leaf in the type object.

These tests would be better served by comparing against the pattern. `dyn.Path` does not make sense for a type object.